### ISSUE TITLE
SQLAlchemy Declared Attribute: comparison-with-callable

### ISSUE THREAD

**Issue by vestronge** [PyCQA/pylint#3955]
### Steps to reproduce
1. app.py
```
from flask import Flask
from flask_sqlalchemy import SQLAlchemy

app = Flask(__name__)
app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///:memory:'
db = SQLAlchemy(app)
```

2. models.py
```
from sqlalchemy import Column, Integer, ForeignKey
from sqlalchemy.ext.declarative import declared_attr

from app import db

class Transaction(db.Model):
   
    @declared_attr
    def account_id(cls):
        return Column(Integer, ForeignKey("account.id", ondelete='CASCADE'))

    @classmethod
    def get_transaction_ids_for_account(cls, account_id):
        return {x[0] for x in cls.query.with_entities('transaction_id').filter(cls.account_id == account_id).all()}  # comparison-with-callable
```

### Current behavior
Last line commented with `comparison-with-callable` raises pylint error `comparison-with-callable`

### Expected behavior
There shouldn't be any error

### pylint --version output
2.6.0

**Comment by hippo91** [comment]
@vestronge thanks for your report. I don't know `SQLAlchemy` at all, but i think that the `declared_attr` turns the decorated method into a kind of class property. Am i right? 
After investigating a bit i think that `pylint` is unable to understand the role of the `declared_attr` decorator. The following code doesn't trigger any message emission, which proves that `pylint` is able to deal with property inside comparison statement.
```
#pylint: disable=missing-module-docstring,missing-class-docstring
#pylint: disable=missing-function-docstring
class Transaction:
    @property
    def account_id(self):
        return True

    def get_transaction_ids_for_account(self, account_id):
        return self.account_id == account_id
```
The `astroid` brain ('brain_sqlalchemy.py') needs probably enhancement.

**Comment by vestronge** [comment]
Hi. The `declared_attr` is essentially declaring a foreign key relationship. And yes, SQLAlchemy has a weird syntax. `.filter(cls.account_id == account_id)` is actually not comparing `cls.account_id` to `account_id` (as we'd normally expect `==` operator to do), but it is creating a SQL statement of the below form 

Example:
```
cls.query.with_entities('transaction_id').filter(cls.account_id == 6).all()
```

translates to

```
select transaction_id from transaction where account_id=6
```

**Comment by hippo91** [comment]
@vestronge thanks for those explanations.

**Comment by AntonOfTheWoods** [comment]
I am also faced with this. @hippo91 is the consensus that this should be closed as "wontfix" and pushed to `sqlalchemy`? Or is there a solution for pylint to play nice with current `sqlalchemy`? Or a fix wanted?

**Comment by hippo91** [comment]
@AntonOfTheWoods can you, please, give me a clear snipped that illustrates the problem and also the python version you use and all of the dependencies (flask version, sqlalchemy version ...)?

**Comment by AntonOfTheWoods** [comment]
@hippo91 , sorry for dropping the ball on this... too many useless github notifications

```
(testpylint) anton@zub:~/dev/tmp/testpylint$ python --version && pip freeze
Python 3.9.7
astroid==2.8.0
greenlet==1.1.1
isort==5.9.3
lazy-object-proxy==1.6.0
mccabe==0.6.1
platformdirs==2.3.0
pylint==2.11.1
SQLAlchemy==1.4.23
toml==0.10.2
typing-extensions==3.10.0.2
wrapt==1.12.1
```
`pylintwarning.py`
```
# pylint: disable=C0321 C0114 C0115 R0903 C0116 R0903
from sqlalchemy.ext.declarative import declared_attr
from sqlalchemy import Column
from sqlalchemy.sql.schema import ForeignKey
from sqlalchemy.future import select


class Timestampit():
    @declared_attr
    def created_by_id(cls):  # pylint: disable=E0213
        return Column("created_by_id", ForeignKey("authuser.id"), nullable=False)


USER_ID = 1
select(Timestampit).where(Timestampit.created_by_id == USER_ID,)  # here is the W0143 issue
```
`$ pylint pylintwarning.py`
```
************* Module pylintwarning
pylintwarning.py:15:26: W0143: Comparing against a callable, did you omit the parenthesis? (comparison-with-callable)

------------------------------------------------------------------
Your code has been rated at 8.89/10 (previous run: 6.36/10, +2.53)
```
The `SQLAlchemy` doc for the functionality: https://docs.sqlalchemy.org/en/13/orm/extensions/declarative/api.html#sqlalchemy.ext.declarative.declared_attr

Is that sufficient?
